app/src/main/java/com/naman14/timberx/sdl/SdlService.kt

```diff
@@ -229,8 +245,14 @@ class SdlService : Service(), KoinComponent {
         }
     }
 
-    private fun onNowPlaying(nowPlaying: Bundle) {
+    private fun onNowPlaying(nowPlaying: Bundle?) {
+        if(nowPlaying == null || sdlManager == null) {
+            return
+        }
         val metadata = nowPlaying.getParcelable<MediaMetadataCompat>(Constants.SONG_METADATA)
+        if(metadata == null) {
+            return;
+        }
```

Comment:
```suggestion
    private fun onNowPlaying(nowPlaying: Bundle) {
        val metadata = nowPlaying.getParcelable<MediaMetadataCompat>(Constants.SONG_METADATA)
        if (metadata == null) {
            return;
        }
```
First check shouldn't be needed after my other suggestion